Tools/_sunrise/pr_check_summary.py

```
@@ -0,0 +1,726 @@
+                page_items = data or []
+
+            results.extend(page_items)
+            if len(page_items) < params["per_page"]:
+                break
+            page += 1
+
+        return results
+
+    def download_job_log(self, job_id: int) -> str:
+        url = f"{GITHUB_API_URL}/repos/{self.repository}/actions/jobs/{job_id}/logs"
+        request = urllib.request.Request(url, method="GET", headers=self.headers("application/vnd.github+json"))
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)